File: src/main/java/no/nav/sosialhjelp/soknad/business/db/repositories/oppgave/OppgaveRepository.java
//package no.nav.sosialhjelp.soknad.business.db.repositories.oppgave;
//
//import no.nav.sosialhjelp.soknad.business.batch.oppgave.Oppgave;
//
//import java.util.Map;
//import java.util.Optional;
//
//public interface OppgaveRepository {
//
//    void opprett(Oppgave oppgave);
//
//    Optional<Oppgave> hentNeste();
//
//    Optional<Oppgave> hentOppgave(String behandlingsId);
//
//    void oppdater(Oppgave oppgave);
//
//    Map<String, Integer> hentStatus();
//
//    int retryOppgaveStuckUnderArbeid();
//
//    void slettOppgave(String behandlingsId);
//}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/db/repositories/oppgave/OppgaveRepository.kt
package no.nav.sosialhjelp.soknad.db.repositories.oppgave

interface OppgaveRepository {
    fun opprett(oppgave: Oppgave)
    fun hentNeste(): Oppgave?
    fun hentOppgave(behandlingsId: String): Oppgave?
    fun hentOppgaveIdList(behandlingsIdList: List<String>): List<Long>
    fun oppdater(oppgave: Oppgave)
    fun hentAntallFeilede(): Int
    fun hentAntallStuckUnderArbeid(): Int
    fun retryOppgaveStuckUnderArbeid(): Int
    fun slettOppgaver(oppgaveIdList: List<Long>)
    fun count(): Int
}
